pkg/config/node.go

```diff
@@ -313,17 +321,30 @@ func GetLBConfig(kubeconfigPath string, apiPort, lbPort, statPort uint16, apiVip
 		LbPort:   lbPort,
 		StatPort: statPort,
 	}
+
 	// LB frontend address: IPv6 '::' , IPv4 ''
 	if apiVip.To4() == nil {
 		config.FrontendAddr = "::"
 	}
-
-	backends, err := getSortedBackends(kubeconfigPath)
+	// since:
+	// 1: Keepalived script for Api vip checks HAProxy LB status in addition to localhost-kube-apiserver.
+	// 2: Masters nodes details needed for HAProxy config retrieved from kube-apiserver .
+	// to break the circular dependency we should first read masters details from localhost:kube-apiserver instead of
+	// api-vip:kube-apiserver.
+	// Since master nodes details are used (in unicast mode) in api VRRP instance at keeplived config
+	// we should first retrieve the information from localhost:kube-apiserver (to avoid circular dependency),
+	// and only failover to api-vip:kube-apiserver
+	backends, err := getSortedBackends("")
```

Comment:
This feels backward to me. Other than the initial deployment where the VIP isn't up yet, the normal case is going to be that the API VIP is responding and we want to use that. If we check localhost first, then every time the local apiserver goes down for any reason we'll spam the logs with these messages. I think I'd prefer to do the VIP check first and fall back to localhost only if it isn't available. What do you think?

Reply:
Sounds like a good idea if we only need to handle HAProxy backends details retrieval, but since this code will be used also for retrieving keepalived API_VIP group members (unicast mode), I'm not sure it's a good idea to have that circular dependency.
I can change the order per your recommendation for this PR and will update this in the unicast PR.